static/scripts/view.js

```diff
@@ -789,18 +789,160 @@ function checkFormValidity(event) {
+var hasShownSubmissionMessage = false;
+var isSubmittingChanges = false;
+
+function checkBugChange(bugid, lastMod, inMsg) {
+  var tiqitApi = new TiqitApi();
+  var msg = inMsg;
+
+  // Prepare the async event functions
+  bugChangeOnLoad = function(eventLoad) {
+    var submitChanges = true;
+
+    if (eventLoad.target.status == 200 && eventLoad.target.responseXML) {
+      // Check if the first date predates lastMod and only Note or Attachment
+      // related changes were made.
+      // If not, we may have overwriteable changes. (See hasHistoryOverwriteableChanges).
+      // If this is the case, check with the user if bug changes should be submitted.
+      var history = historyDataFromXML(eventLoad.target.responseXML);
+      var lastModHistoryDate = new Date(history[0]['Date']);
+      var lastModDate = new Date(lastMod);
+      var hasHistoryOverwriteableChanges = false;
+
+      // Checking for legitimate changes here
+      if (lastModHistoryDate > lastModDate) {
+        for (i in history) {
+          var historyItem = history[i];
+          var historyItemDate = new Date(historyItem['Date']);
+          if (historyItem['Field'] != "File Name" &&
+              historyItem['Field'] != "Note Title" &&
+              historyItemDate > lastModDate) {
+                hasHistoryOverwriteableChanges = true;
+                break;
+          }
+        }
+      }
+
+      // Inform the user of changes that will be overwritten.
+      if (hasHistoryOverwriteableChanges) {
+        msg += "Your bug view is from " + lastModDate.toLocaleString();
+        msg += ", and the bug now contains changes from " + lastModHistoryDate.toLocaleString() + ".\n\n";
+        msg += "The following set of changes to this bug will likely be overwritten:\n\n";
+        for (i in history) {
+          var historyItem = history[i];
+          var historyItemDate = new Date(historyItem['Date']);
+          if (historyItem['Field'] == "File Name" ||
+              historyItem['Field'] == "Note Title") {
+                // Do not tell the user that their note or attachment
+                // may be overwritten. It won't be!
+                continue;
+          }
+          if (historyItemDate > lastModDate) {
+            if (historyItem['Operation'] == 'Delete') {
+              msg += "    Deleted '" + historyItem['Field'] + "', which was set to '" + historyItem['OldValue'] + "'\n";
+            } else if (historyItem['Operation'] == 'New Record') {
+              if (historyItem['NewValue'] == "") {
+                msg += "    Created " + historyItem['Field'] + "\n";
+              } else {
+                msg += "    Created '" + historyItem['Field'] + "' as '" + historyItem['NewValue'] + "'\n";
+              }
+            } else if (historyItem['Operation'] == 'Modify') {
+              if (historyItem['OldValue'] == "" && historyItem['NewValue'] == "") {
+                msg += "    Modified '" + historyItem['Field'] + "'\n";
```

Comment:
Does it make sense for empty values to be treated differently from non-empty values? I haven't seen this running to it's very possible that there's a good reason.

Reply:
This just matches the behaviour when getting history for a bug when in search mode. Only a cosmetic reason really to remove cases where we might see:

Set 'FieldExample' from '' to ''

We can't guarantee that a Tiqit user's bug database/backend will always provide something sensible.